### ISSUE TITLE
installsiteman?dir and installvendorman?dir cannot be set by Configure

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5782]
Migrated from [rt.perl.org#15579](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=15579) (status was 'resolved')

Searchable as RT15579$

**Comment by p5pRT** [comment]
### From utx@penguin.cz

#### Created by sb@utx.cz



Following variables\, used by lib/ExtUtils/MM\_Unix\.pm\, should be\, as
comments in this file says\, customizable by defaults from lib/Config\.pm
by an argument to Configure script\. But these variables are not cited
there&#8203;:

installsiteman1dir
installsiteman3dir
installvendorman1dir
installvendorman3dir

The fix is simple \- add these variables to Configure script\. Their defaults
can be ''\, or if installman?dir is set\, but install\(site|vendor\)man?dir
are not\, they should default to value of installman?dir\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

Site configuration information for perl v5.8.0:

Configured by sb at Wed Jul 24 15:53:13 CEST 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.19-rc1, archname=i686-linux-thread-multi
    uname='linux utx 2.4.19-rc1 #13 st čen 26 00:43:12 cest 2002 i686 unknown '
    config_args='-d -Dprefix=/usr -Duseshrplib -Dinstallman1dir=/usr/share/man/man1 -Dman1dir=/usr/share/man/man1 -Dinstallman3dir=/usr/share/man/man3 -Dman3dir=/usr/share/man/man3 -Dinstallsiteman1dir=/usr/share/man/man1 -Dman1dir=/usr/share/man/man1 -Dinstallsiteman3dir=/usr/share/man/man3 -Dman3dir=/usr/share/man/man3 -Dinstallvendorman1dir=/usr/share/man/man1 -Dman1dir=/usr/share/man/man1 -Dinstallvendorman3dir=/usr/share/man/man3 -Dman3dir=/usr/share/man/man3 -Dcccdlflags=-fPIC -Doptimize=-O3 -fomit-frame-pointer  -Dusethreads -Duseithreads -Dusedevel -Dinstallusrbinperl -Uversiononly'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr//include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3 -fomit-frame-pointer ',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr//include'
    ccversion='', gccversion='3.1', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldbm -ldb -ldl -lm -lpthread -lc -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
    libc=/lib/libc-2.2.5.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.2.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/usr/lib/perl5/5.8.0/i686-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /usr/lib/perl5/5.8.0/i686-linux-thread-multi
    /usr/lib/perl5/5.8.0
    /usr/lib/perl5/site_perl/5.8.0/i686-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/sb
    LANG=cs_CZ.ISO-8859-2
    LANGUAGE=cs_CZ.ISO-8859-2:sk_SK.ISO-8859-2
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/bin:/usr/bin:/usr/games:/usr/X11R6/bin:/usr/tex/bin:/opt/gnome/bin:/opt/gcc-2.95/bin:/opt/enlightenment/bin:/opt/media/bin:/opt/netpbm/bin:/opt/qt3/bin:/opt/kde/bin:/opt/mozilla/bin:/opt/AbiSuite/bin:/opt/dtp/bin:/opt/misc/bin:/opt/misc/games:/opt/3d/bin:/opt/qt2/bin:/opt/j2re1.4.0/bin:/opt/binary/bin:/opt/autostuff-old/bin:/opt/libtool-1.3.5/bin:/opt/gettext-old/bin:/opt/gettext-0.10/bin:/opt/qcad/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From freeweb@nyckelpiga.de

Hallo perl5\-porters\,

$ diff \-urdb perl\-5\.8\.0/Configure~ perl\-5\.8\.0/Configure
<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.0/Configure~       2002-07-19 00:55:15.000000000 +0200
+++ perl-5.8.0/Configure        2002-07-26 11:32:34.000000000 +0200
@@ -917,10 +917,14 @@
 malloctype=''
 usemymalloc=''
 installman1dir=''
+installsiteman1dir=''
+installsiteman1dir=''
 man1dir=''
 man1direxp=''
 man1ext=''
 installman3dir=''
+installsiteman3dir=''
+installvendorman3dir=''
 man3dir=''
 man3direxp=''
 man3ext=''



```
</details>
Gerrit
\-\- 
$ make signature
make&#8203;: \*\*\* No rule to make target \`signature'\.  Stop\.

**Comment by p5pRT** [comment]
### From freeweb@nyckelpiga.de

Guten Tag Gerrit P\. Haase\,

am Freitag\, 26\. Juli 2002 um 11&#8203;:37 schrieben Sie&#8203;:

> Hallo perl5\-porters\,

Oops\, should read&#8203;:
<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.0/Configure~       2002-07-19 00:55:15.000000000 +0200
+++ perl-5.8.0/Configure        2002-07-26 11:32:34.000000000 +0200
@@ -917,10 +917,14 @@
 malloctype=''
 usemymalloc=''
 installman1dir=''
+installsiteman1dir=''
+installvendorman1dir=''
 man1dir=''
 man1direxp=''
 man1ext=''
 installman3dir=''
+installsiteman3dir=''
+installvendorman3dir=''
 man3dir=''
 man3direxp=''
 man3ext=''




```
</details>
> $ diff -urdb perl-5.8.0/Configure~ perl-5.8.0/Configure
> \-\-\- perl\-5\.8\.0/Configure~       2002\-07\-19 00&#8203;:55&#8203;:15\.000000000 \+0200
> \+\+\+ perl\-5\.8\.0/Configure        2002\-07\-26 11&#8203;:32&#8203;:34\.000000000 \+0200
> @&#8203;@&#8203; \-917\,10 \+917\,14 @&#8203;@&#8203;
>  malloctype=''
>  usemymalloc=''
>  installman1dir=''
> \+installsiteman1dir=''
> \+installsiteman1dir=''

Sorry\,
\-\- 
&nbsp;                   \_\) |  
&nbsp;  \_\` |  \-\_\)  \_|  \_| |  \_|
\\\_\_\, |\\\_\_\_|\_|  \_|  \_|\\\_\_|
\_\_\_\_/

**Comment by p5pRT** [comment]
### From @doughera88

On 25 Jul 2002\, Stanislav Brabec wrote&#8203;:

> Following variables\, used by lib/ExtUtils/MM\_Unix\.pm\, should be\, as
> comments in this file says\, customizable by defaults from lib/Config\.pm
> by an argument to Configure script\. But these variables are not cited
> there&#8203;:
> 
> installsiteman1dir
> installsiteman3dir
> installvendorman1dir
> installvendorman3dir
> 
> The fix is simple \- add these variables to Configure script\. Their defaults
> can be ''\, or if installman?dir is set\, but install\(site|vendor\)man?dir
> are not\, they should default to value of installman?dir\.

See the thread starting at
http&#8203;://www\.xray\.mpe\.mpg\.de/mailing\-lists/perl5\-porters/2002\-07/msg01319\.html

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu
&nbsp;    Dept\. of Physics
&nbsp;    Lafayette College\, Easton PA 18042

**Comment by p5pRT** [comment]
### From @jhi

This issue has been now fixed \(perl change \#17951\)\, and the fix will be in Perl 5\.8\.1\, whenever that happens\.  I'm marking the problem ticket as resolved\.

**Comment by p5pRT** [comment]
@jhi - Status changed from 'new' to 'resolved'
